torch_xla/csrc/helpers.cpp

```diff
@@ -578,6 +700,137 @@ std::pair<xla::XlaOp, xla::XlaOp> XlaHelpers::PromoteSecond(xla::XlaOp op1,
+      if (broadcast_sizes[i] != kUnboundedSize) {
+        get_dim_ops.push_back(
+            XlaHelpers::ScalarValue<int32_t>(broadcast_sizes[i], op.builder()));
+
+        auto s = ShapeHelper::ShapeOfXlaOp(get_dim_ops.back());
+        std::cout << "implicitB shape: " << xla::ShapeUtil::HumanString(s)
+                  << " for size: " << broadcast_sizes[i] << "\n";
+      } else {
+        get_dim_ops.push_back(xla::GetDimensionSize(op, i));
+
+        auto s = ShapeHelper::ShapeOfXlaOp(get_dim_ops.back());
+        std::cout << "implicitB shape: " << xla::ShapeUtil::HumanString(s)
+                  << " for size: ? of index: " << i << "\n";
+      }
+    }
+  }
+
+  size_t min_size = std::min(op_shape_dims.size(), aux_shape_dims.size());
+  for (int i = 0; i < min_size; i++) {
+    int op_shape_index = op_shape_dims.size() - min_size + i;
+    int aux_op_shape_index = aux_shape_dims.size() - min_size + i;
+    int shape_index = shape_dims.size() - min_size + i;
+
+    int64_t op_shape_dim = op_shape_dims[op_shape_index];
+    int64_t aux_op_shape_dim = aux_shape_dims[aux_op_shape_index];
+    int64_t shape_dim = shape_dims[shape_index];
+
+    // op_shape   aux_op_shape   shape
+    //   1            X            X
+    //   X            1            X
+    //   X            X            X
+    //   1            ?            ? (from aux_op_shape)
+    //   ?            1            ? (from op_shape)
+    //   X            ?            X
+    //   ?            X            X
+    //   ?            ?            ? (from any, let's select op_shape)
+    // where X != kUnboundedSize && X != 1
+    if (shape_dim != kUnboundedSize) {
+      get_dim_ops.push_back(ScalarValue<int32_t>(shape_dim, op.builder()));
+
+      auto s = ShapeHelper::ShapeOfXlaOp(get_dim_ops.back());
+      std::cout << "implicitB shape: " << xla::ShapeUtil::HumanString(s)
+                << " for size: " << shape_dim << "\n";
+
+    } else if (op_shape_dim == 1 || aux_op_shape_dim == 1) {
+      if (op_shape_dim == 1) {
+        get_dim_ops.push_back(
+            xla::GetDimensionSize(aux_op, aux_op_shape_index));
+
+        auto s = ShapeHelper::ShapeOfXlaOp(get_dim_ops.back());
+        std::cout << "implicitB shape: " << xla::ShapeUtil::HumanString(s)
+                  << " for size: ? of index: " << aux_op_shape_index << "\n";
+
+      } else {
+        get_dim_ops.push_back(xla::GetDimensionSize(op, op_shape_index));
+
+        auto s = ShapeHelper::ShapeOfXlaOp(get_dim_ops.back());
+        std::cout << "implicitB shape: " << xla::ShapeUtil::HumanString(s)
+                  << " for size: ? of index: " << op_shape_index << "\n";
+      }
+    } else {
+      get_dim_ops.push_back(xla::GetDimensionSize(op, op_shape_index));
+
+      auto s = ShapeHelper::ShapeOfXlaOp(get_dim_ops.back());
+      std::cout << "implicitB shape: " << xla::ShapeUtil::HumanString(s)
+                << " for size: ? of index: " << op_shape_index << "\n";
```

Comment:
all of these seems like debug string added instead of meaningful VLOG, I am not sure if we want to keep these.. if we do than give it like `VLOG(7)` so it won't be too noisy